website/components/Modals/ShareModal.tsx

```diff
@@ -3,27 +3,29 @@ import { ShareAltOutlined } from "@ant-design/icons";
 import { Button, Input, Modal, notification } from "antd";
 import React, { useRef, useState } from "react";
 import styled from "styled-components";
+import { useShallow } from "zustand/shallow";
 
 import { MultisceneUrls } from "../../../src/aics-image-viewer/components/App/types";
-import { ViewerStateContextType } from "../../../src/aics-image-viewer/components/ViewerStateProvider/types";
+import { selectViewerSettings, useViewerState, ViewerStore } from "../../../src/aics-image-viewer/state/store";
 import { AppDataProps } from "../../types";
 import { ENCODED_COLON_REGEX, ENCODED_COMMA_REGEX, serializeViewerUrlParams } from "../../utils/url_utils";
 import { FlexRow } from "../LandingPage/utils";
 
-import {
-  ALL_VIEWER_STATE_KEYS,
-  connectToViewerState,
-} from "../../../src/aics-image-viewer/components/ViewerStateProvider";
-
 type ShareModalProps = {
   appProps: AppDataProps;
   // Used to retrieve the current camera position information
   view3dRef?: React.RefObject<View3d | null>;
-} & ViewerStateContextType;
+};
 
 const ModalContainer = styled.div``;
 
 const ShareModal: React.FC<ShareModalProps> = (props: ShareModalProps) => {
+  const selectViewerSettingsShallow = useShallow(selectViewerSettings);
+  const selectChannelSettingsShallow = useShallow((store: ViewerStore) => store.channelSettings);
+
+  const viewerSettings = useViewerState(selectViewerSettingsShallow);
+  const channelSettings = useViewerState(selectChannelSettingsShallow);
```

Comment:
Would it make sense to make these one-liners if `selectViewerSettingsShallow` is not being used anywhere? e.g.
`const viewerSettings = useViewerState(useShallow(selectViewerSettings))`

Reply:
Yeah probably! Just felt weird having a hook in a hook like that.